src/main/java/nextstep/subway/favorite/domain/Favorite.java

```diff
@@ -0,0 +1,84 @@
+package nextstep.subway.favorite.domain;
+
+import nextstep.subway.BaseEntity;
+import nextstep.subway.enums.ErrorMessage;
+import nextstep.subway.member.domain.Member;
+import nextstep.subway.station.domain.Station;
+
+import javax.persistence.*;
+import java.util.Objects;
+
+@Entity
+public class Favorite extends BaseEntity {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @ManyToOne
+    @JoinColumn(name = "member_id", foreignKey = @ForeignKey(name = "fk_favorite_to_member"), nullable = false)
+    private Member member;
+    @ManyToOne
+    @JoinColumn(name = "source_station_id", foreignKey = @ForeignKey(name = "fk_favorite_to_sourcestation"), nullable = false)
+    private Station sourceStation;
+    @ManyToOne
+    @JoinColumn(name = "target_station_id", foreignKey = @ForeignKey(name = "fk_favorite_to_targetstation"), nullable = false)
+    private Station targetStation;
+
+    protected Favorite() {
+    }
+
+    private Favorite(Member member, Station sourceStation, Station targetStation) {
+        validateSourceAndTargetEquals(sourceStation, targetStation);
+        this.member = member;
+        this.sourceStation = sourceStation;
+        this.targetStation = targetStation;
+    }
+
+    public static Favorite of(Member member, Station sourceStation, Station targetStation) {
+        return new Favorite(member, sourceStation, targetStation);
+    }
+
+    private void validateSourceAndTargetEquals(Station sourceStation, Station targetStation) {
+        if(sourceStation.equals(targetStation)) {
```

Comment:
```suggestion
        if (sourceStation.equals(targetStation)) {
```

전체적으로 각 파일에서 띄어쓰기가 잘못되어있는 부분들이 간혹 보이네요, ide 에서 제공하는 코드 프리티를 전체적으로 적용해보는것도 의미있을 것 같습니다 👍 

`띄어쓰기에 대한 컨벤션 규칙이 모호함`